sherpa/bin/streaming_conformer_rnnt/streaming_server.py

```diff
@@ -0,0 +1,632 @@
+#!/usr/bin/env python3
+# Copyright      2022  Xiaomi Corp.        (authors: Fangjun Kuang,
+#                                                    Wei Kang)
+#
+# See LICENSE for clarification regarding multiple authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A server for streaming ASR recognition. By streaming it means the audio samples
+are coming in real-time. You don't need to wait until all audio samples are
+captured before sending them for recognition.
+
+It supports multiple clients sending at the same time.
+
+Usage:
+    ./streaming_server.py --help
+
+    ./streaming_server.py
+"""
+
+import argparse
+import asyncio
+import http
+import logging
+import math
+import warnings
+from concurrent.futures import ThreadPoolExecutor
+from typing import List, Optional, Tuple
+
+import numpy as np
+import sentencepiece as spm
+import torch
+import websockets
+from sherpa import RnntConformerModel, streaming_greedy_search
+
+from decode import Stream
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    parser.add_argument(
+        "--port",
+        type=int,
+        default=6006,
+        help="The server will listen on this port",
+    )
+
+    parser.add_argument(
+        "--nn-model-filename",
+        type=str,
+        help="""The torchscript model. You can use
+          icefall/egs/librispeech/ASR/pruned_transducer_statelessX/export.py \
```

Comment:
Yes, I prefer to modify the code in the existed folders, it would be more clear to see what are the difference between streaming and non-streaming models.  You guys might have different opinions though.

Reply:
I think in the current icefall, whenever we change something, we create a new folder for it. 

It is ok as long as it is backward compatible.